packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-walkthrough.ts

```diff
@@ -42,8 +44,14 @@ export async function serviceWalkthrough(context: $TSContext, datasourceMetadata
 
   const { inputs, availableRegions } = datasourceMetadata;
 
+  // FIXME: We should NOT be treating CloudFormation templates as inputs to prompts! This a temporary exception while we move team-provider-info to a service.
```

Comment:
could we please move this part into a function. like 
.. application state ..
const execMode = context.getIsModeInteractive() // check mode is -y or --headless
const template = get<name>CFNTemplate() 

.. business logic...
const selectedRegion = getSelectedRegion( template, execMode). 

Imo it's safer to not rely on cloudformation input to infer execution state. It would be great if we can get this state from  a single signal.

Reply:
We're not inferring the execution state here, just bypassing the prompt if the necessary data is already available. The code here is not aware of execution state.

Reply:
the prompt itself will fail in headless mode if the value can't be derived from available info (or prompt if mode is not headless). this is handled by the prompter library.

Reply:
I see, so the code is agnostic to the execution mode and the prompter returns the error-message for API failures of non-interactive calls . 
This is okay for now and we can make the change as part of the bigger CLI refactor: 

With the current implementation we may face problems when we scale the application beyond the current implementation to say add more states of execution or different input modes. 
Imo, The Prompter library should be state agnostic.  However business-logic is better state-aware.
`Client=>HeadlessAPI-Call=> ValidateRequestAgainstState  => WalkthroughInHeadlessMode => Prompter`
                                                     (api failure is handled here)
- what we have is
`Client=>HeadlessAPI-Call=> WalkthroughModeAgnostic => Prompter=>Validate

Reply:
Agreed - the good news is we do have plan to resolve this with @edwardfoyle's work on extracting this metadata into a service. Once that is complete we can remove this (but it may be a few months, and customers are in a broken state today).